server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -364,7 +364,12 @@ public static Injector makeInjectorWithModules(final Injector baseInjector, Iter
       extensionModules.addModule(module);
     }
 
-    return Guice.createInjector(Modules.override(intermediateModules).with(extensionModules.getModules()));
+    return Guice.createInjector(
+        Modules.combine(
+            Modules.override(intermediateModules).with(extensionModules.getModules()),
+            new FunctionInitializer()
```

Comment:
Does this need to be down here instead of in the base modules list? It feels weird to have one module that is "more special" than all the others.